### PR TITLE
 Bipartite Graph 

### PR DESCRIPTION
**Pull request by Mrjoy832**
## Related Issue
 #188  

## Proposed Changes
-Added short description about problem statement
-Added code, related input-output

## Checklist

- [X] ✅ My code follows the code style of this project.
- [X] 📝 My change requires a change to the documentation.
- [X] 🎀 I have updated the documentation accordingly.
- [X] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [X] ✨ I have added tests to cover my changes.
- [X] 🚩 All new and existing tests passed.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] 🌟 ed the repo

**Solution of issue #188** 
Added brief description about the **bipartite graph** and explained a code using BFS algorithm 
Here I used 2 inputs for the 1st one I show that the graph is bipartite and 2nd input graph is not bipartite
**APPROACH::**
It is possible to test whether a graph is bipartite or not using a BFS Algorithm
There are 2 ways to check for a bipartite graph:
**1. A graph is a bipartite graph if and only if it is colorable by 2 colors only.**
While doing BFS traversal m the given Graph, each node in the BFS tree is given its parent’s opposite color. If there exists an edge connecting the current vertex to a previously colored vertex with the same color, then we can say that the graph is not bipartite.

**2. A graph is a bipartite graph if and only if it  contains an odd cycle(Cycle contains  Odd number of vertices)**
If a graph contains an odd, we cannot divide the graph such that every adjacent vertex has a different color.  If in the BFS, we find an edge, if both of its endpoints are at the same level, then the graph is not Bipartite and an odd-cycle is found. 

**Here in the explanation, we use the 1st approach of the BFS algorithm to check whether a given graph I bipartite or not.**

## Output Screenshots
   

| INPUT-1
bipartite1
 | INPUT-2   |
bipartite2

### CODE REVIEW

**Comment by Mrjoy832** [comment]
@miraj0507  please check it...it was already reviewd but I deleted PR by mastake ..I had messaged you on discord about it ..please check it .

**Comment by Mrjoy832** [comment]
@Lakhankumawat  what should I have to  change here ?

**Review by siddhi-244 — CHANGES_REQUESTED** [review]
I can see you have committed 3 files ! Also you added code in cpp file, you don't need it in readme. Follow proper structure of Readme

**Comment by Mrjoy832** [comment]
> I can see you have committed 3 files ! Also you added code in cpp file, you don't need it in readme. Follow proper structure of Readme 

Ok I will update

**Comment by Mrjoy832** [comment]
image
How to solve it? @siddhi-244

**Comment by Mrjoy832** [comment]
@siddhi-244  , updated please check

**Comment by siddhi-244** [comment]
Check the tasks which are completed by you !

**Comment by Mrjoy832** [comment]
> Check the tasks which are completed by you !

@siddhi-244 , done please check

**Review by siddhi-244 — APPROVED** [review]
LGTM

File: G-Graphs/README.md [hunk 1]
```diff
@@ -42,47 +42,8 @@ If a graph contains an odd, we cannot divide the graph such that every adjacent
 <br>//O(N+E) for adjacency List , O(N) for color array, O(N) for queue.
 ---
```
Comment (Lakhankumawat) [inline G-Graphs/README.md:43]: make a README inside B-BipartiteGraph and add your README there , remove your README edits from outside

**Review by Lakhankumawat — CHANGES_REQUESTED** [review]

**Comment by Mrjoy832** [comment]
@Lakhankumawat , updated

**Comment by Lakhankumawat** [comment]
@all-contributors please add @Mrjoy832 ,

**Comment by Lakhankumawat** [comment]
@all-contributors please add @Mrjoy832 code ,

**Pull request merged**